src/compiler/checker.ts

```diff
@@ -91,6 +93,9 @@ module ts {
         var globalBooleanType: ObjectType;
         var globalRegExpType: ObjectType;
         var globalTemplateStringsArrayType: ObjectType;
+        var globalPromiseType: ObjectType;
+        var globalIPromiseType: ObjectType;

```

Comment:
I can change the name here, but I chose the name as its looking for an `IPromise` interface in lib.d.ts.  This is to be consistent with the `IArguments` interface. For clarity and to disambiguate, I’ll use `globalPromiseLikeInterfaceType` so there is less confusion between this type and the `globalPromiseType` type.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 3:44 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24632686>:

> @@ -91,6 +93,9 @@ module ts {

>          var globalBooleanType: ObjectType;

>          var globalRegExpType: ObjectType;

>          var globalTemplateStringsArrayType: ObjectType;

> +        var globalPromiseType: ObjectType;

> +        var globalIPromiseType: ObjectType;

The "I" prefix feels like more of a C# convention. Maybe globalPromiseInterface or something like that

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24632686>.